mlir/test/Catalyst/AnnotateDebugAsEnzymeConst.mlir

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// RUN: quantum-opt %s --annotate-debug-callback-as-enzyme-const --split-input-file --verify-diagnostics | FileCheck %s
+
+// This test just makes sure that we can
+// run the compiler with the option
+//
+//   --annotate-debug-callback-as-enzyme-const
+// and that if there are no callbacks present
+// it doesn't change anything
+
+// CHECK-LABEL: @foo
+func.func @foo() {
+  // CHECK-NOT: llvm.mlir.global external @__enzyme_inactivefn
```

Comment:
Can a global appear inside the scope of a function?

Reply:
I don't think so, but just because the `// CHECK-NOT` is inside the function does not mean that the code is inside the function. I added a more strict test here: https://github.com/PennyLaneAI/catalyst/pull/706/commits/1c43d479c55c4c2b5fe5b46dc2d511cb1bcb4323